### ISSUE TITLE
Use better error messages/handle fatal errors more gracefully

### ISSUE THREAD

**Issue by robinpaulson** [martykan/forecastie#316]
At present, a number of problems are represented with a small set of error messages, which are at best misleading. We will replace these with a set which are more representative and which hence assist the user to diagnose the problem. A number of issues contain these bugs; they are:
#324
#292 
#258 
#251 
#234 
#460

#349 Has an unhandled error

**Comment by robinpaulson** [comment]
The error messages are here, in strings.xml
https://github.com/martykan/forecastie/blob/master/app/src/main/res/values/strings.xml#L91

**Comment by FridoDeluxe** [comment]
I noticed another thing concerning the error messages which I assume is wrong. The `GenericRequestTask` receives the response from OWM and creates an error message if necessary. But one text is misleading as far as I can judge.

There might be the case that there is a response from OWM, but it is neither successfull (HTTP 200) nor one of the other known error causes. So it's most probably another client or server error (HTTP 4xx and 5xx), but the shown error message will be _"There is a problem with your internet connection."_ (`msg_connection_problem`). I think it should rather state that something is wrong with the request or response.

There exists already an error message that covers network problems/`IOException`: _"Connection not available."_ (`msg_connection_not_available`).

**Comment by robinpaulson** [comment]
> I think it should rather state that something is wrong with the request or response

Yes, I agree.

**Comment by FridoDeluxe** [comment]
What do you think about: _"Error while requesting data."_ ? Something like _"Please try again."_ most likely makes no sense in this case.

**Comment by robinpaulson** [comment]
Error while requesting or processing data?

Or is that too much information for a simple error message?

**Comment by FridoDeluxe** [comment]
No, not too much information, I slightly changed it to _Error while requesting or receiving data_: #480.
